Autopilot/PathManager/Src/pathStateClasses.cpp

```diff
@@ -120,4 +164,287 @@ pathManagerState& fatalFailureMode::getInstance()
     return singleton;
 }
 
+/****************************************************************************************************
+
+LANDING STATE FUNCTIONS
+
+****************************************************************************************************/
+
+void landingTransitionStage::execute(pathManager* pathMgr)
+{
+    if(!pathMgr -> madeLandingPoints)
+    {
+        //requires data structure that dhruv wants to use 
+        landingPath = WaypointManager(0.0,0.0);
+        path = LandingManager::createSlopeWaypoints(getFromTelemetry::telemetryInput, sensorFusion::sensorInput.altitude);
+
+        //creating waypoints 
+        pathArray[0] = landingPath.initialize_waypoint(path.intersectionPoint.longitude, path.intersectionPoint.latitude, path.intersectionPoint.altitude, PATH_FOLLOW);
+        pathArray[1] = landingPath.initialize_waypoint(path.aimingPoint.longitude, path.aimingPoint.latitude, path.aimingPoint.altitude, PATH_FOLLOW);
+        pathArray[2] = landingPath.initialize_waypoint(path.stoppingPoint.longitude, path.stoppingPoint.latitude, path.stoppingPoint.altitude, PATH_FOLLOW);
+        currentLocation = landingPath.initialize_waypoint(sensorFusion::sensorInput.longitude, sensorFusion::sensorInput.latitude, sensorFusion::sensorInput.altitude, HOLD_WAYPOINT); //fill in with sensor fusion data
+        //initializing flight path
+        waypointStatus = landingPath.initialize_flight_path(pathArray, 3, currentLocation);
+        
+        //set made madelandingPoints to true
+        pathMgr->madeLandingPoints = true;
+    }
+
+    //follow the landing waypoints
+    waypointStatus = landingPath.get_next_directions(sensorFusion::sensorInput, &cruisingState::_outputdata);
+
+    //calculating the difference in heading to detect if finished turning (2 differences in heading possible)
+    differenceInHeading1 = getFromTelemetry::telemetryInput.stoppingDirectionHeading - sensorFusion::sensorInput.track;
+    differenceInHeading2 = sensorFusion::sensorInput.track - getFromTelemetry::telemetryInput.stoppingDirectionHeading;
```

Comment:
I don't like how we are just accessing variables from other classes like this. I would rather you made getter functions since it will prevent people (in the future) from actually changing the values.

Reply:
Answered above